### ISSUE TITLE
show quoted identifiers with quotation marks in Database Navigator

### ISSUE THREAD

**Issue by wistlo** [dbeaver/dbeaver#21764]
For Postgres connections. the Database Navigator shows Columns with names and data type but gives no indication as to whether or not the identifier is quoted with " or the case of each character. 

I suggest showing Column identifiers as quoted and with their true case, if that is how they are stored in the schema (not specific to Postgres, though Postgres in its default configuration is more sensitive about exact matches on case and quoted identifiers than other DBs, such as mySQL).  

Alternative solution is to access **Generate Sql > DDL**, which pops up a window showing the statement creating the identifier as it was originally declared with case and quoted attributes.  This extra step is a minor inconvenience for the user, so this suggestion would not be a high priority.  

If this does get implemented, it could be offered as an option where users who prefer not to see the "ugly" quoted identifiers can turn it off. 

For those of us required to use quoted identifiers to show names as exactly as they are defined by external sources (honoring case, spaces, hyphens, parens, etc.)   this would be a helpful feature especially when dealing with tables made up of dozens of fields.

**Comment by E1izabeth** [comment]
We don't understand the idea and the use case. Could you show example of how does your table names look like in some other tool?
For example, if you're creating the table `tAbLe` in database where identifiers are stored in lower-case, then table with name `table` will be created unless you're using quotes `"tAbLe"`. What do you want us to show in this case? Do you want to see this `"tAbLe"` in navigator with quotes?

**Comment by wistlo** [comment]
Yes, I would want an option to see **"tAble"** in navigator with quotes.  This would apply to columns and any other objects where a quoted identifier can be used.   With the **Generate SQL->DDL** preview, we can see what we need to.  I suggest this as a low priority, as it would make checking object names just a little faster and easier by showing up in their actual quoted/unquoted format in the Database Navigator.

Don't need another tool to demonstrate this, as dBeaver has the DDL feature.  Use case is when I'm working with a table and need to specify column name identifers precisely.  For example in Database Navigator shows
![image](https://github.com/dbeaver/dbeaver/assets/154039/083164c8-6802-4021-8460-a60942484ff5)
but when I use that table in code, I need to know the exact format of the identifier, which is always shown correctly in DDL:
![image](https://github.com/dbeaver/dbeaver/assets/154039/0c9ecd63-3b8b-44c7-99fd-df9476cc6e18)

Same applies to tables, for example:
![image](https://github.com/dbeaver/dbeaver/assets/154039/41159669-d172-4b85-82ea-89dc47a71273)
is actually this in code:
![image](https://github.com/dbeaver/dbeaver/assets/154039/e82cca1a-3954-4468-90ab-cb3f143912c1)
The quoted table is the exact outside filename from source as specified in the outside source design and as it is delivered to our system.  Anything we derive or generate internally is in lowercase_underscore unquoted format.

We have tables that are mix of quoted fields from outside sources and internal fields we create as part of our process, like the integer count at the top here:
![image](https://github.com/dbeaver/dbeaver/assets/154039/c946913c-9007-43d6-a26a-9d9fcf9c5946)
The formatting is not obvious if using Database Navigator:
![image](https://github.com/dbeaver/dbeaver/assets/154039/2ccf5ad8-ada8-4584-b597-6f40e4c74886)


We don't have any examples outside of tables and columns, but any object (view, function, etc.) where a quoted identifier is allowed would be subject to this suggestion.  

————————

You may be wondering "Why? Why would anyone choose this?"  The suggestion is of no use on projects where object names are not quoted, as is best practice on a greenfield project.  

We chose to use the identifiers so that fields coming in from different sources can immediately be identified by their quirky name on a spreadsheet or CSV such as **"a1( idclass, legacy)"**—a real example— and not translated to, say, **a1_idclass_legacy**.  We also have fields coming in with names like **SELECT**, **PRIMARY**, **NAME**, etc. that are actual Postgres keywords, produced by sources where adaptation to a database was never under consideration.  Rather than adapt the names to some sort of convention, we decided to just quote them directly.  With hundreds of tables from outside sources, each with dozens of fields, this took away the need to remap object names to something different.  Just quote them and go.  

Not sure it was the best design choice, but Postgres allows it and we went that way.  dBeaver is 3/4 of the way there already, with DDL preview and the quotation respect shown by autocomplete on quoted fields ([it was just fixed in 23.2.4](https://github.com/dbeaver/dbeaver/issues/21589), in fact).

I can imagine many users would be horrified to see quotes showing up uninvited in the Database Navigator (though case already does), so this should be an option, not a fixed feature.  Again, not a high priority, but would be nice to have for the few of us working with object names such as "**Name (as registered ®)**" (another real example).

**Comment by E1izabeth** [comment]
I see. Thank you for detailed explanation ☺️

**Comment by LonwoLonwo** [comment]
I added the wait-for-votes label for other users:
You can vote for the issue with the emoji under the issue description to promote it to a higher priority.
